### DISCUSSION TITLE
Upon exiting a modal on a record page, the record page values should be updated

### DISCUSSION THREAD

**Discussion by skamensky** [cortezaproject/corteza-server#783] [Ideas]
When modifying child record in a modal, there may be workflows that update the parent. In order for the user to see the updated values on the page, they currently have to refrresh it.

**Comment by Fajfa** [comment]
If I understand correctly, this would apply if you were on a record page and edited a related record by opening it in a modal.

So would this apply to record lists as well? Let's say you edit a record in a modal and save it, should the record list that contains that record also be re-fetched?

**Reply by skamensky** [reply]
You understood correct. 

Yes, the record list should also be refetched, similar to the "Refreshes all record values on the page" prompt step in workflows. All data on the page should be refreshed, in my opinion. If something is displayed on the page and a user triggered modification occured, there's a high likelyhood that the user modified data is an input to some other data's automated field calculation on the page.

**Reply by skamensky** [reply]
A workaround could be to add a "Block auto-refresh interval" option for a record type block. 

![image](https://user-images.githubusercontent.com/19151369/212646673-5e333703-a218-4271-99e4-d39c82e1d2c8.png)

**Reply by Fajfa** [reply]
A workaround would be that, but I agree it would make sense to refresh the related data. 
In the case of a record page, it's usually easier since it's only one resource (a record) that the page blocks are in relation to. 
A non-record page though can have blocks that are related to the resource that has been modified or not. 
So knowing which page blocks to refresh, is the main thing that should be discussed.

**Reply by skamensky** [reply]
This might be a bit heavy-handed, but adding some kind of limited user definable Trigger->Action logic might be a solution.

For example, modules X,Y, and Z are present on the page. When Y gets modified/updated refresh Z's data.
The trigger is modified/update and the action is refresh data.

This could be scaled to a much more robust low-code UI builder framework, but I think to start it would serve a lot of use cases.

Another idea (though I think would be a more tedious user experience) is to create the same user definable Trigger->Action logic but on the component level (e.g. if Component1 gets modified/updated, refresh Component2's data).

**Reply by skamensky** [reply]
In any case if you need to support generic pages, and not just record pages, I think some level of user defined refresh logic needs to be supported.

What do you think?
